packages/investor-foxy/src/api/api.ts

```diff
@@ -0,0 +1,856 @@
+      userAddress,
+      type,
+      wallet
+    } = input
+    this.verifyAddresses([userAddress, contractAddress])
+    if (!wallet) throw new Error('Missing inputs')
+
+    let estimatedGasBN: BigNumber
+    try {
+      estimatedGasBN = await this.estimateWithdrawGas(input)
+    } catch (e) {
+      throw new Error(`Estimate Gas Error: ${e}`)
+    }
+
+    const stakingContract = this.foxyStakingContracts.find(
+      (item) => toLower(item.options.address) === toLower(contractAddress)
+    )
+    if (!stakingContract) throw new Error('Not a valid contract address')
+
+    const isDelayed = type === WithdrawType.DELAYED && amountDesired
+    if (isDelayed && !amountDesired.gt(0)) throw new Error('Must send valid amount')
+
+    const data: string = isDelayed
+      ? stakingContract.methods.unstake(amountDesired.toString(), true).encodeABI({
+          from: userAddress
+        })
+      : stakingContract.methods.instantUnstake(true).encodeABI({
+          from: userAddress
+        })
+
+    let nonce: number
+    try {
+      nonce = await this.web3.eth.getTransactionCount(userAddress)
+    } catch (e) {
+      throw new Error(`Get nonce Error: ${e}`)
+    }
+    let gasPrice: string
+    try {
+      gasPrice = await this.web3.eth.getGasPrice()
+    } catch (e) {
+      throw new Error(`Get gasPrice Error: ${e}`)
+    }
+    const estimatedGas = estimatedGasBN.toString()
+    const bip44Params = this.adapter.buildBIP44Params({ accountNumber })
+    const chainId = Number(this.network)
+    const payload = {
+      bip44Params,
+      chainId,
+      data,
+      estimatedGas,
+      gasPrice,
+      nonce: String(nonce),
+      to: contractAddress,
+      value: '0'
+    }
+    const txToSign = buildTxToSign(payload)
+    if (wallet.supportsOfflineSigning()) {
+      const signedTx = await this.adapter.signTransaction({ txToSign, wallet })
+      if (dryRun) return signedTx
+      return this.broadcastTx(signedTx)
+    } else if (wallet.supportsBroadcast() && this.adapter.signAndBroadcastTransaction) {
+      if (dryRun) {
+        throw new Error(`Cannot perform a dry run with wallet of type ${wallet.getVendor()}`)
+      }
+      return this.adapter.signAndBroadcastTransaction({ txToSign, wallet })
+    } else {
+      throw new Error('Invalid HDWallet configuration ')
+    }
+  }
+
+  async claimWithdraw(input: ClaimWithdrawal): Promise<string> {
+    const {
+      accountNumber = 0,
+      dryRun = false,
+      contractAddress,
+      userAddress,
+      claimAddress,
+      wallet
+    } = input
+    const addressToClaim = claimAddress ?? userAddress
+    this.verifyAddresses([userAddress, contractAddress, addressToClaim])
+    if (!wallet) throw new Error('Missing inputs')
+
+    let estimatedGasBN: BigNumber
+    try {
+      estimatedGasBN = await this.estimateClaimWithdrawGas(input)
+    } catch (e) {
+      throw new Error(`Estimate Gas Error: ${e}`)
+    }
+
+    const stakingContract = this.foxyStakingContracts.find(
```

Comment:
anywhere `this.foxyStakingContracts.find` is used we should be calling something like `this.getStakingContract` that throws if it can't find one